### PR TITLE
Integrate the qwen series models via dashscope-sdk.

### PR DESCRIPTION
**Pull request by jiangsier-xyz**
Qwen series models are provided by Alibaba Cloud. They are much better in Asia languages then other LLMs.

DashScope is a model service platform. Qwen models are its primary supported models. But it also supports other series like LLaMA2, Dolly, ChatGLM, BiLLa(based on LLaMA)...These may be integrated sometime in the future.

### CODE REVIEW

File: langchain4j-parent/pom.xml [hunk 1]
```diff
@@ -127,6 +127,12 @@
                 <artifactId>snakeyaml</artifactId>
                 <version>2.0</version>
             </dependency>
+
+            <dependency>
+                <groupId>com.alibaba</groupId>
+                <artifactId>dashscope-sdk-java</artifactId>
```
Comment (langchain4j) [inline langchain4j-parent/pom.xml]: Could you please move all Qwen-related code to a separate `langchain4j-qwen` module? Same like `langchain4j-pinecone`, `langchain4j-weaviate`, etc.

**Review by ghost — CHANGES_REQUESTED** [review]
Hi @jiangsier-xyz thanks a lot for your contribution! All good, the only thing: please move it to a separate module. Thanks!

**Comment by langchain4j** [comment]
@jiangsier-xyz BTW what would be the best name for the module and classes? Qwen* or DashScope* or ModelScope*? Taking into consideration that other non-Qwen models might be added later.

File: langchain4j-parent/pom.xml (hunk 1 above)
Comment (jiangsier-xyz) [reply langchain4j-parent/pom.xml]: OK, I will move the implementation to a separate module.
About the module name, "qwen" is a name of model family (just like "gpt-3.5/4..." in openai), and dashscope is a model service provider (just like openai?). Furthermore, modelscope is a model community (just like huggingface).
Should I use langchain4j-dashscope here?

**Comment by jiangsier-xyz** [comment]
> @jiangsier-xyz BTW what would be the best name for the module and classes? Qwen* or DashScope* or ModelScope*? Taking into consideration that other non-Qwen models might be added later.

According to me, we might design the package name based on the "model service provider", and the class name for model family. For example, let's imagine tha openai introduced a model series different from gpt one day, then how to refactor OpenAiChatModel?

For now, dashscope has a unified client, but it take "QwenParam" for qwen model family. It take different parameters for other models. For example, LLaMa2 served by dashscope needs less options than qwen. It's a better way to design a LlaMaChatModel in the same package than mix it into a QwenChatModel/DashScopeChatModel.

What do you think?

**Comment by langchain4j** [comment]
@jiangsier-xyz I hope that model providers will have a unified interface that will work for most of the models, so that we and them don't have to maintain dozens of APIs/classes for each single model family. This is the case for OpenAI and HuggingFace (at least so far). In the case of QwenParam, I agree, better to keep it isolated then. Then `langchain4j-dashscope` and `Qwen*Model` class names, right?

**Comment by jiangsier-xyz** [comment]
> @jiangsier-xyz I hope that model providers will have a unified interface that will work for most of the models, so that we and them don't have to maintain dozens of APIs/classes for each single model family. This is the case for OpenAI and HuggingFace (at least so far). In the case of QwenParam, I agree, better to keep it isolated then. Then `langchain4j-dashscope` and `Qwen*Model` class names, right?

Yeap. Thanks for the feedback. I will refactor the module in the near future.

**Review by ghost — DISMISSED** [review]
Thanks a lot @jiangsier-xyz! Great job!

**Review by kuraleta — APPROVED** [review]

**Pull request merged**